### PR TITLE
Seek Improvements

### PR DESCRIPTION
**Pull request by garyrussell**
- add convenience methods to `AbstractConsumerSeekAware`

- compile tests with Java 11

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -133,6 +133,10 @@ subprojects { subproject ->
 		targetCompatibility = 1.8
 	}
 
+	compileTestJava {
+		sourceCompatibility = 11
```
Comment (artembilan) [inline build.gradle:137]: What made you to move to Java 11?
Thanks

Comment (garyrussell) [reply build.gradle:137]: The main driver was `Map.of(new TopicPartition("foo", 0), 0L, new TopicPartition("foo", 1), 0L)` in tests. I hate building maps in tests with `put()`; that's why I added `MapBuilder` in SI and SA (but we don't have one in SK).

But we also get new language features (`var`) - like we did with SI 4.3.x and using Java 8 in tests.

Comment (artembilan) [reply build.gradle:137]: Sure! Just curious what could be so useful in other projects.

We might need re-test the whole project to be sure that JavaDocs, for example, are built OK, too.

Comment (garyrussell) [reply build.gradle:137]: I actually went to 14 initially, but travis only goes up to 11 😦

Comment (artembilan) [reply build.gradle:137]: Good to know. Merging...

Comment (garyrussell) [reply build.gradle:137]: Ugh... https://stackoverflow.com/questions/51992347/missing-iframe-view-for-javadocs-jdk-11

**Comment by artembilan** [comment]
```
24-Aug-2020 17:29:57 | > Task :api
-- | --
24-Aug-2020 17:29:57 | 1 error
24-Aug-2020 17:29:57 | src/api/overview.html:4: error: self-closing element not allowed
24-Aug-2020 17:29:57 | <hr/>
24-Aug-2020 17:29:57 | ^
24-Aug-2020 17:29:57 |  
24-Aug-2020 17:29:57 | > Task :api FAILED
```
Hm. I should've gave you a couple minutes to check it before merging upstream.

Anyway let try switch it to Java 14 on CI!

File: build.gradle (hunk 1 above)
Comment (garyrussell) [reply build.gradle:137]: I guess we'll get used to it.

**Pull request merged**